[CI] RecoveryDuringReplicationTests.testResyncAfterPrimaryPromotion failing reproducibly

This test just failed and reproduces locally for me.

https://gradle-enterprise.elastic.co/s/lflqfnox2xl4c/tests/nfxodyych4uas-b7kbbqymo3ykc

```
    java.lang.AssertionError: 
    Expected: is <5>
         but: was <1>
        at __randomizedtesting.SeedInfo.seed([F5E9DF78AE4335DB:F82474D3A9A0FC1F]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.replication.RecoveryDuringReplicationTests.testResyncAfterPrimaryPromotion(RecoveryDuringReplicationTests.java:380)
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.replication.RecoveryDuringReplicationTests.testResyncAfterPrimaryPromotion" -Dtests.seed=F5E9DF78AE4335DB -Dtests.security.manager=true -Dtests.locale=es-ES -Dtests.timezone=Europe/Uzhgorod -Dcompiler.java=13
```

--- a/server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java
+++ b/server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java
@@ -74,6 +74,7 @@ import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
 import static org.hamcrest.Matchers.anyOf;
+import static org.hamcrest.Matchers.either;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.everyItem;
@@ -309,9 +310,7 @@ public class RecoveryDuringReplicationTests extends ESIndexLevelReplicationTestC
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/52598")
     public void testResyncAfterPrimaryPromotion() throws Exception {
-        // TODO: check translog trimming functionality once rollback is implemented in Lucene (ES trimming is done)
         String mappings = "{ \"_doc\": { \"properties\": { \"f\": { \"type\": \"keyword\"} }}}";
         try (ReplicationGroup shards = new ReplicationGroup(buildIndexMetaData(2, mappings))) {
             shards.startAll();
@@ -378,7 +377,7 @@ public class RecoveryDuringReplicationTests extends ESIndexLevelReplicationTestC
                     assertThat(source.source.utf8ToString(), is("{ \"f\": \"normal\"}"));
                 }
             }
-            assertThat(translogOperations, is(initialDocs + extraDocs));
+            assertThat(translogOperations, either(equalTo(initialDocs + extraDocs)).or(equalTo(task.getResyncedOperations())));
         }
     }
 

